CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs

```diff
@@ -0,0 +1,253 @@
+                Margin = new Thickness(0, 0, 8, 0)
+            });
+            loadingMsgPanel.AddElementToStackPanel(new TextBlock
+            {
+                Text = "Loading the Image",
+                FontWeight = FontWeights.SemiBold
+            });
+
+            parentGrid.AddElementToGridRowColumn(imageCropper, 0, 0);
+            parentGrid.AddElementToGridRowColumn(loadingMsgPanel, 0, 0);
+
+            StorageFile file = await StorageFile.GetFileFromPathAsync(filePath);
+            await imageCropper.LoadImageFromFile(file);
+
+            GC.WaitForPendingFinalizers();
+            GC.WaitForFullGCComplete();
+
+            Storyboard storyboardAnim = new Storyboard();
+            DoubleAnimation loadingMsgPanelAnim = loadingMsgPanel.CreateDoubleAnimation("Opacity", 0, 1, null, TimeSpan.FromMilliseconds(500), EasingType.Cubic.ToEasingFunction());
+            DoubleAnimation imageCropperAnim = imageCropper.CreateDoubleAnimation("Opacity", 1, 0, null, TimeSpan.FromMilliseconds(500), EasingType.Cubic.ToEasingFunction());
+            storyboardAnim.Children.Add(loadingMsgPanelAnim);
+            storyboardAnim.Children.Add(imageCropperAnim);
+            storyboardAnim.Begin();
+
+            dialogOverlay.IsPrimaryButtonEnabled = true;
+        }
+
+        private static async Task<FileStream> GenerateCachedStream(FileInfo InputFileInfo, uint ToWidth, uint ToHeight, bool isFromCropProcess = false)
+        {
+            FileStream cachedFileStream;
+            if (isFromCropProcess)
+            {
+                string InputFileName = InputFileInfo.FullName;
+                InputFileInfo.MoveTo(InputFileInfo.FullName + "_old", true);
+                FileInfo newCachedFileInfo = new FileInfo(InputFileName);
+
+                cachedFileStream = newCachedFileInfo.Create();
+                using (FileStream oldInputFileStream = InputFileInfo.OpenRead())
+                {
+                    await ResizeImageStream(oldInputFileStream, cachedFileStream, ToWidth, ToHeight);
+                    cachedFileStream.Position = 0;
+                }
+                InputFileInfo.Delete();
+                return cachedFileStream;
+            }
+
+            FileInfo cachedFileInfo = GetCacheFileInfo(InputFileInfo.FullName + InputFileInfo.Length);
+            bool isCachedFileExist = cachedFileInfo.Exists && cachedFileInfo.Length > 1 << 15;
+            if (isCachedFileExist) return cachedFileInfo.OpenRead();
+
+            cachedFileStream = cachedFileInfo.Create();
+            using (FileStream inputFileStream = InputFileInfo.OpenRead())
+            {
+                await ResizeImageStream(inputFileStream, cachedFileStream, ToWidth, ToHeight);
+                cachedFileStream.Position = 0;
+            }
+            return cachedFileStream;
+        }
+
+        internal static FileInfo GetCacheFileInfo(string filePath)
+        {
+            string cachedFileHash = ConverterTool.BytesToCRC32Simple(filePath);
```

Comment:
Since we use MD5 on pretty much everything, is there particular reason why we use CRC32 here?

Reply:
the reason is why we use CRC32 is that, we need to calculate the unique hash to have minimum impact on performance. We have another alternative like XXH64 if we need to consider to change it.

Reply:
Ah ok, all good we can keep it as is for now